setup.py

```diff
@@ -4,6 +4,7 @@
     'numpy',
     'scipy',
     'scikit-learn',
+    'matplotlib'
```

Comment:
Why do we need a dependency of matplotlib? If matplotlib is only used in the example, I prefer not including it to keep the package light.